runbooks/how-to-create-server-regression-dashboards.md

```diff
@@ -0,0 +1,343 @@
+|      Analysis       |                                       A work-in-progress Dashboard that uses a Dataset to display fields                                       |
+|      Dashboard      |                                          A collection of graphs and visuals that represents a Dataset                                          |
+
+## FAQ
+
+<!-- This section should contain frequently-asked-questions as sub-headers with answers as sub-paragraphs. Since this is a living document, this section should be added to as needed. -->
+
+<!-- REPLACE ME when FAQs are added -->
+
+## Instructions
+
+1. You will need to navigate to the QuickSight AWS Service in order to start creating QuickSight Dashboards
+
+   1. Navigate to `aws.amazon.com` in any browser
+   2. Click the "Sign In" link in the navigation bar on the top right of the screen
+   3. Enter your AWS account credentials and sign-in
+   4. You should now have navigated to the "Console Home", click the search input in the navigation bar on the top left of the screen
+   5. Type "QuickSight" and click the QuickSight result that appears
+
+2. Every QuickSight dashboard requires a _Dataset_, which for the `server-regression` dashboards will be a `Custom SQL` query against Athena. As there will be three dashboards, there will be three datasets (one dataset for each dashboard)
+
+   1. Creating the `bfd-test-server-regression` dataset:
+
+      1. Navigate to "Datasets" in QuickSight (click on "Datasets" in the navigation tree on the left side of the page)
+      2. Click the "New dataset" button on the top-left of the page
+      3. The page that appears should list a large number of possible data sources. Find "Athena" in this list, and click on it
+      4. Enter "bfd-test-server-regression" for the "Data source name"
+      5. Choose `bfd` for the "Athena workgroup"
+      6. Click "Create data source"
+      7. Choose `AwsDataCatalog` as the "Catalog"
+      8. Choose `bfd-insights-bfd-test` as the "Database"
+      9. Choose `bfd_insights_bfd_test_server_regression" as the "Table"
+      10. Click "Use custom SQL"
+      11. In the first input box, enter "bfd-test-server-regression" (replacing "New custom SQL")
+      12. In the large text area, enter the following SQL query:
+
+      ```sql
+      select
+        totals.total_reqs_per_second,
+        totals.num_requests as total_requests,
+        totals.num_failures as total_failures,
+        totals.min_response_time as total_min_response_time,
+        totals.average_response_time as total_avg_response_time,
+        totals.median_response_time as total_median_response_time,
+        totals.response_time_percentiles."0.95" as total_95_percentile,
+        totals.response_time_percentiles."0.99" as total_99_percentile,
+        totals.response_time_percentiles."1.0" as total_100_percentile,
+        metadata.compare_result,
+        date_format(from_unixtime(metadata.timestamp, 'US/Eastern'), '%b-%e-%y %T EST') as date_time,
```

Comment:
Also is the test data always recorded in eastern? (I guess the s3 instances/lambdas are running on us-east, so that is a relatively safe assumption, unless we have a disaster scenario and need to recover on west)

Reply:
I chose EST as much of our team is based around EST. This was definitely not due to any limitations, `metadata.timestamp` is a Unix time timestamp, meaning that we could use any timezone (as Unix time is based on UTC).

To be fair, I'm not entirely sure what QuickSight supports as far as display time. I'll check if there's a way to change the timezone depending on the viewer's timezone. We could also determine if a different timezone is more suitable (i.e. UTC).

Reply:
After checking, there _are_ [workarounds](https://community.amazonquicksight.com/t/time-zone-adjustments/652) for timezone and date handling in QuickSight, but none of them are automatic based upon the viewer's timezone. I honestly think the value-to-effort ratio here is way higher than just defaulting to a particular timezone, as the runbook _and_ each Analysis and Dataset in QuickSight would need to be updated to support this. 

For now, I still feel that keeping it to EST works well _enough_ our use-case.